### PR TITLE
feat(tile-converter): estimation of time remaining

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: modules/tile-converter/src/i3s-converter/helpers/progress.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import process from 'process';
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/helpers/progress.ts:1]: Nit: You are add more hard Node.js dependencies by importing `process`. In general I prefer making code as portable as possible, even if your current usage is node only. I think some of this converter code might be useful in the browser at some point, perhaps for on-the-fly conversion. 

Also having to deal with bigints for timing makes no sense, no user is going to care about nano-second precision.

There is a portable utility is probe.gl {getHighResTimestamp} that you could use: https://github.com/uber-web/probe.gl/blob/master/modules/log/src/index.ts#L15.

You could import it or copy it.

probe.gl is designed to help measure apps, and if some API you need is missing we could add it.

File: modules/tile-converter/src/i3s-converter/helpers/progress.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    this.nanoSecForOneItem = nanoSecForOneItem;
+
+    const nanoSecInSec = BigInt(1e9);
+    const timeRemainingInSeconds =
+      (BigInt(this._stepsTotal - this._stepsDone) * nanoSecForOneItem) / nanoSecInSec;
+    return {timeRemaining: Number(timeRemainingInSeconds), trust: trust};
+  }
+
+  /**
+   * Check if the computed velociy of the process can be considered trust.
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/helpers/progress.ts:118]: What does this mean? Whether we can believe the numbers we get? Maybe rephrase so it is easier to understand?

**Review by ibgreen — APPROVED** [review]

File: modules/tile-converter/src/i3s-converter/helpers/progress.ts (hunk 1 above)
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/helpers/progress.ts:1]: Bigint issue is fixed. Will consider using probe.gl. Thanks for the hint.

File: modules/tile-converter/src/i3s-converter/helpers/progress.ts (hunk 2 above)
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/helpers/progress.ts:116]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 3]
```diff
@@ -641,6 +622,25 @@ export default class I3SConverter {
...
+      if (timeRemainingObjectBasedOnCount !== null && timeRemainingObjectBasedOnCount.trust) {
+        timeRemainingString = `${timeRemainingObjectBasedOnCount.timeRemaining.toFixed(0)} seconds`;
+      }
+
+      let percentString = this.progresses[PROGRESS_PHASE1_COUNT].getPercentString();
+      if (percentString) {
+        percentString = ' ' + percentString;
+      }
+      console.log(
+        `[converted${percentString}%, time remaining = ${timeRemainingString}]: ${sourceTile.id}`
```
Comment (belom88) [inline modules/tile-converter/src/i3s-converter/i3s-converter.ts:641]: As discussed:
```suggestion
        `[converted${percentString}%, ${timeRemainingString} left]: ${sourceTile.id}`
```

**Review by belom88 — CHANGES_REQUESTED** [review]
Add basic tests for the new file

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 4]
```diff
@@ -641,6 +622,25 @@ export default class I3SConverter {
       transform: transformationMatrix,
       parentNodes: childNodes
     };
+
+    if (sourceTile.id) {
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts:626]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts [hunk 5]
```diff
@@ -641,6 +622,25 @@ export default class I3SConverter {
...
+      this.progresses[PROGRESS_PHASE1_COUNT].stepsDone += 1;
+
+      let timeRemainingString = 'Calculating...';
+      const timeRemainingObjectBasedOnCount =
+        this.progresses[PROGRESS_PHASE1_COUNT].getTimeRemaining();
+      if (timeRemainingObjectBasedOnCount !== null && timeRemainingObjectBasedOnCount.trust) {
+        timeRemainingString = `${timeRemainingObjectBasedOnCount.timeRemaining.toFixed(0)} seconds`;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts:633]: Done

File: modules/tile-converter/src/i3s-converter/i3s-converter.ts (hunk 3 above)
Comment (mspivak-actionengine) [reply modules/tile-converter/src/i3s-converter/i3s-converter.ts:641]: Done, but removed source.id from the "converted" string

**Review by belom88 — APPROVED** [review]

**Review by belom88 — APPROVED** [review]
